### PR TITLE
	Remove various references to 7.0 and 7.1 transport versions

### PR DESCRIPTION
**Pull request by thecoop**
Remove references to 7.0 and 7.1 TVs for cleanup of those versions

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/TransportVersions.java [hunk 1]
```diff
@@ -53,7 +53,6 @@ static TransportVersion def(int id) {
     @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA) // remove the transport versions with which v9 will not need to interact
     public static final TransportVersion ZERO = def(0);
     public static final TransportVersion V_7_0_0 = def(7_00_00_99);
```
Comment (mosche) [inline server/src/main/java/org/elasticsearch/TransportVersions.java:55]: Are you keeping 7.0 here?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/mapper/ExpressionRoleMapping.java [hunk 2]
```diff
@@ -139,11 +138,7 @@ public ExpressionRoleMapping(StreamInput in) throws IOException {
         this.name = in.readString();
         this.enabled = in.readBoolean();
         this.roles = in.readStringCollectionAsList();
-        if (in.getTransportVersion().onOrAfter(TransportVersions.V_7_2_0)) {
```
Comment (mosche) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/mapper/ExpressionRoleMapping.java:142]: Leaked into the wrong PR?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/mapper/ExpressionRoleMapping.java [hunk 3]
```diff
@@ -175,9 +170,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(name);
         out.writeBoolean(enabled);
         out.writeStringCollection(roles);
-        if (out.getTransportVersion().onOrAfter(TransportVersions.V_7_2_0)) {
```
Comment (mosche) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/mapper/ExpressionRoleMapping.java:178]: same

**Review by mosche — APPROVED** [review]
LGTM, just two minor points... most of all, why not remove the constant for 7.0?

File: server/src/main/java/org/elasticsearch/TransportVersions.java (hunk 1 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/TransportVersions.java:55]: There's still some references to this in various tests

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/mapper/ExpressionRoleMapping.java (hunk 2 above)
Comment (thecoop) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/mapper/ExpressionRoleMapping.java:142]: Yes, but eeeeh, it'll be going anyway...

**Pull request merged**